vulcanai/models/metrics.py

```diff
@@ -903,11 +904,57 @@ def _boot_cv(network, data_loader, k, epochs, retain_graph, valid_interv,
                 "\n\n***KeyboardInterrupt: Cross validate stopped \
                 prematurely.***\n\n")
 
+    @staticmethod
+    def stratified_split(dataset, k, strata_column="class_label"):
+        """ Split data into k subsets evenly over the given strata_column
+        Does not work for greater than 2-D data.
+
+        Parameters:
+            dataset : data.dataset
+                The dataset to be split.
+            k : int
+                The number of folds to split the training into.
+            strata_column: string or int
+                Either "class_label" or integer index of column.
+                Default "class_label"
+
+        Returns:
+            split_datasets : list
+            A list of dataset objects split into k parts.
+
+        """
+
+        if strata_column == "class_label":
+            sr = pd.Series(dataset.tensors[1].numpy())
+        else:
+            # just let it fail if it doesn't work
+            # does not work for complex multidimensional data
+            sr = pd.Series(dataset.tensors[0][strata_column].numpy())
```

Comment:
sorry, why are you taking the 0-ith vs 1-st index of the tensors? and to clarify we will let it fail if the num of dims is >2?

Reply:
because the 0th is the data and the 1st is the target. I've added comments to that effect. 
and, yes, we will let it fail.

Reply:
any way to generalize to multiple target tensors?

Reply:
I think that's out of scope of this function and something that isn't supported in any of our current vulcan implementation, unless I'm mistaken

Reply:
fair enough - you're right